src/main/java/com/google/sps/data/Coordinates.java

```diff
@@ -70,4 +72,34 @@ public double getLatitude() {
     public double getLongitude() {
         return this.longitude;
     }
+
+    /**
+    * Checks whether the latitude and longitude are valid numbers.
+    * @return a boolean representing whether the coordinates are valid.
+    */
+    public boolean isValid() {
+        return -90 <= this.latitude && this.latitude <= 90 && -180 <= this.longitude && this.longitude <= 180;
+    }
+
+    /**
+    * Get a random valid coordinates.
+    * @return a random Coordinates object.
+    */
+    public static Coordinates getRandomCoordinates() {
+        return new Coordinates(-90+Math.random()*180, -180+Math.random()*360);
+    }
+
+    /**
+    * Get random coordinates whose latitude and longitude values are at most d away from this.
+    * If these coordinates are (x,y), then this will return a point whose x is between x-d and x+d,
+    * and whose y is between y-d and y+d.
+    * @param d the upper bound on the difference between the coordinates.
+    */
+    public Coordinates getRandomWithin(double d) {
```

Comment:
Please be more specific for what 'd' is, maybe  "differenceForUpperBound"?

Reply:
Thanks, renamed to "maxDelta".